libtenzir/builtins/commands/rebuild.cpp

```diff
@@ -231,15 +229,16 @@ struct rebuilder_state {
       for (auto&& rp : std::exchange(run->stop_requests, {})) {
         rp.deliver();
       }
-      run.reset();
       if (run->options.detached) {
+        run.reset();
         return;
       }
       if (err) {
         rp.deliver(std::move(err));
         return;
```

Comment:
Why not reset here too?

Reply:
I pushed that roughly ~5s before you submitted your review :D